nixos/modules/services/networking/adguardhome.nix

```diff
@@ -0,0 +1,97 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.adguardhome;
+
+  args = concatStringsSep " " ([
```

Comment:
Are these options not permitted in the configuration file?

Reply:
Yes, but this allows them to be set the first time the service is run. It would also allow the addition of an `openFirewall` option, at least for the web interface. Creating a stub config file if you want a different port or host on the first run is a pain - the program will fail if many options aren't specified, in my opinion it's best left to manage and generated the config file itself.

In the case that the command line and config file disagree, the command like takes precedence and will overwrite the values in the config file.

Reply:
Ah, understood.